src/USVPlugin.cpp

```diff
@@ -1,32 +1,35 @@
 #include <gazebo_usv/USVPlugin.hpp>
-#include "Utilities.hpp"
 
 using namespace std;
 using namespace gazebo;
 using namespace gazebo_usv;
 
+sdf::ElementPtr dispatchPluginIfMatches(
+    sdf::ElementPtr plugin_sdf, std::string const& expectedName
+) {
+    auto pluginName = plugin_sdf->Get<std::string>("name");
+
+    auto match = expectedName == pluginName;
+    if (!match)
+    {
+        return nullptr;
+    }
```

Comment:
```
if (expectedName != pluginName)
{
    return nullptr;
}
```

Reply:
this would not compile, we would have to wrap the comparison anyway: `if (!(expected == actual))`, so I prefer to actually "tag" it under a variable

Reply:
How the hell it doesnt compile? Arent both expectedName and pluginName `std::string`?

Reply:
ahahauhaua probably operator! does not see it as multiple arguments if they are not enclosed into `()`

Reply:
> this would not compile

???

`std::string != std::string` does not compile ?

Reply:
shoot, I was thinking in `if(!std::string==std::string)`

sorry for the noise